pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -201,30 +237,42 @@ func (tc *NoExecuteTaintManager) Run(stopCh <-chan struct{}) {
 				break
 			}
 			podUpdate := item.(*podUpdateItem)
+			hash := hash(podUpdate.nodeName())
 			select {
 			case <-stopCh:
 				break
-			case tc.podUpdateChannel <- podUpdate:
+			case tc.podUpdateChannels[hash%workers] <- podUpdate:
 			}
 		}
 	}(stopCh)
 
+	wg := sync.WaitGroup{}
+	wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go tc.worker(i, wg.Done, stopCh)
+	}
+	wg.Wait()
+}
+
+func (tc *NoExecuteTaintManager) worker(worker int, done func(), stopCh <-chan struct{}) {
+	defer done()
+
 	// When processing events we want to prioritize Node updates over Pod updates,
 	// as NodeUpdates that interest NoExecuteTaintManager should be handled as soon as possible -
 	// we don't want user (or system) to wait until PodUpdate queue is drained before it can
 	// start evicting Pods from tainted Nodes.
 	for {
 		select {
 		case <-stopCh:
-			break
-		case nodeUpdate := <-tc.nodeUpdateChannel:
+			return
```

Comment:
That BTW is a fix for unit tests to make them use negligible amount of CPU instead of 10+ cores :)